File: backend_dotnet/fruit_api/fruit_api/Models/Voucher.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("Vouchers")]
public class Voucher
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("voucherId")]
    public string VoucherId { get; set; } = null!;

    [Required]
    [MaxLength(50)]
    [Column("voucherCode")]
    public string VoucherCode { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    [Column("discountType")]
    public string DiscountType { get; set; } = string.Empty;

    [Required]
    [Column("discountValue", TypeName = "decimal(12,2)")]
    public decimal DiscountValue { get; set; }

    [Column("minOrderValue", TypeName = "decimal(12,2)")]
    public decimal MinOrderValue { get; set; } = 0;

    [Column("maxDiscountValue", TypeName = "decimal(12,2)")]
    public decimal? MaxDiscountValue { get; set; }

    [Required]
    [Column("quantity")]
    public int Quantity { get; set; }

    [Column("usedQuantity")]
    public int UsedQuantity { get; set; } = 0;

    [Column("startDate")]
    public DateTime? StartDate { get; set; }

    [Column("endDate")]
    public DateTime? EndDate { get; set; }

    [MaxLength(20)]
    [Column("status")]
    public string Status { get; set; } = "active";

    // Navigation properties
    public OrderVoucher? OrderVoucher { get; set; }

    public ICollection<UserVoucher>? UserVouchers { get; set; }
}

File: frontend_flutter/web_admin - Copy/lib/models/voucher.dart
class Voucher {
  final String voucherId;
  final String voucherCode;
  final String discountType;
  final double discountValue;
  final double minOrderValue;
  final double? maxDiscountValue;
  final int quantity;
  final int usedQuantity;
  final DateTime? startDate;
  final DateTime? endDate;
  final String status;
  final bool isValid;

  Voucher({
    required this.voucherId,
    required this.voucherCode,
    required this.discountType,
    required this.discountValue,
    required this.minOrderValue,
    this.maxDiscountValue,
    required this.quantity,
    required this.usedQuantity,
    this.startDate,
    this.endDate,
    required this.status,
    required this.isValid,
  });

  factory Voucher.fromJson(Map<String, dynamic> json) {
    return Voucher(
      voucherId: json['voucherId'] ?? '',
      voucherCode: json['voucherCode'] ?? '',
      discountType: json['discountType'] ?? '',
      discountValue: (json['discountValue'] as num).toDouble(),
      minOrderValue: (json['minOrderValue'] as num).toDouble(),
      maxDiscountValue: json['maxDiscountValue'] != null 
          ? (json['maxDiscountValue'] as num).toDouble() 
          : null,
      quantity: json['quantity'] ?? 0,
      usedQuantity: json['usedQuantity'] ?? 0,
      startDate: json['startDate'] != null ? DateTime.parse(json['startDate']) : null,
      endDate: json['endDate'] != null ? DateTime.parse(json['endDate']) : null,
      status: json['status'] ?? 'inactive',
      isValid: json['isValid'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'voucherCode': voucherCode,
      'discountType': discountType,
      'discountValue': discountValue,
      'minOrderValue': minOrderValue,
      'maxDiscountValue': maxDiscountValue,
      'quantity': quantity,
      'startDate': startDate?.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'status': status,
    };
  }
}